src/textual/widgets/_tree.py

```diff
@@ -825,6 +825,8 @@ def _new_id(self) -> NodeID:
         return NodeID(id)
 
     def _get_node(self, line: int) -> TreeNode[TreeDataType] | None:
+        if line < 0:
```

Comment:
This would at least pass existing tests and seems to work when running your MRE:

```suggestion
        if line < 0 and not self.show_root:
```

Reply:
It indeed only happens when show_root=False as otherwise the root is always selected